server/src/nodeinfo.rs

```diff
@@ -0,0 +1,47 @@
+use crate::db::community_view::SiteView;
+use crate::db::establish_connection;
+use crate::version;
+use crate::Settings;
+use actix_web::HttpResponse;
+use actix_web::body::Body;
+use serde_json::json;
+
+pub fn node_info_well_known() -> HttpResponse<Body> {
+  let json = json!({
+    "links": {
+      "rel": "http://nodeinfo.diaspora.software/ns/schema/2.0",
+      "href": format!("https://{}/nodeinfo/2.0.json", Settings::get().hostname),
+    }
+  });
+
+  return HttpResponse::Ok()
+    .content_type("application/json")
+    .body(json.to_string());
+}
+
+pub fn node_info() -> HttpResponse<Body> {
+  let conn = establish_connection();
+  let site_view = match SiteView::read(&conn) {
+    Ok(site_view) => site_view,
+    Err(_e) => return HttpResponse::InternalServerError().finish(),
+  };
```

Comment:
Nice, this is how I handle these elsewhere. IMO this should be fine for errors, elsewhere I use a wrapper, in order to know which specific websocket command failed, and to give more detailed error messages, but for these ones, we can worry about more detailed error messages later.

Reply:
Where is that wrapper?